service/metrics.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 LY Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"github.com/kpango/glg"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type Metrics interface {
+	GetOriginLatency() prometheus.Histogram
+}
+
+type metrics struct {
+	httpOriginLatency prometheus.Histogram
+}
+
+func NewMetrics() Metrics {
+	m := &metrics{
+		httpOriginLatency: prometheus.NewHistogram(prometheus.HistogramOpts{
+			Name:    "http_origin_latency_in_seconds",
+			Help:    "Origin latency in seconds",
+			Buckets: prometheus.DefBuckets,
+		}),
+	}
+
+	err := prometheus.Register(m.httpOriginLatency)
+	if err != nil {
+		if registered, ok := err.(prometheus.AlreadyRegisteredError); ok {
+			prometheus.Unregister(registered.ExistingCollector)
+			prometheus.MustRegister(m.httpOriginLatency)
+		} else {
+			glg.Errorf("Failed to register collector: %v", err)
+		}
```

Comment:
1. MustRegister() will panic, remove
2. should fail sidecar startup on register error
```suggestion
		return fmt.Errorf("Failed to register collector: %v", err)
```

Reply:
@WindzCUHK fixedin https://github.com/AthenZ/authorization-proxy/pull/30/commits/1a0bc8cef3624f605e59ddad85deec897964644b